src/D2L.CodeStyle.Analyzers/Immutability/MutabilityInspector.cs

```diff
@@ -138,15 +138,9 @@ HashSet<ITypeSymbol> typeStack
 
 			// If we're not verifying immutability, we might be able to bail out early
 			var scope = type.GetImmutabilityScope();
-			if( !flags.HasFlag( MutabilityInspectionFlags.IgnoreImmutabilityAttribute ) && scope != ImmutabilityScope.None ) {
-				// if we're fully immutable or we allow unsealed (i.e., base classes), then bailout
-				if( scope == ImmutabilityScope.SelfAndChildren ) {
-					ImmutableHashSet<string> immutableExceptions = type.GetAllImmutableExceptions();
-					return MutabilityInspectionResult.NotMutable( immutableExceptions );
-				} else if( scope == ImmutabilityScope.Self && flags.HasFlag( MutabilityInspectionFlags.AllowUnsealed ) ) {

```

Comment:
IIUC from the PR OP, this predicate would only ever be true when getting called from `InspectConcreteType`, so it's moved into `InspectConcreteType` directly (with the modification of also checking `SelfAndChildren` as a short-circuit). Makes sense.

Reply:
Yep!